yamusic/yamusic.go

```diff
@@ -161,21 +168,22 @@ func (c *Client) Do(
 		return nil, err
 	}
 
-	defer func() {
-		closeErr := resp.Body.Close()
-		if closeErr != nil {
-			log.Println("close response body error: ", closeErr)
-		}
-	}()
-
 	if v != nil {
 		if w, ok := v.(io.Writer); ok {
 			_, err = io.Copy(w, resp.Body)
 			if err != nil {
 				return nil, err
 			}
 		} else {
-			err = json.NewDecoder(resp.Body).Decode(v)
+			dat, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
+			}
+			resp.Body = io.NopCloser(bytes.NewReader(dat))
+			if c.Debug {
+				debl.Println("Got empty")
```

Comment:
I'm not sure why it always prints `Got empty`. Isn't it in some cases only?

Reply:
Oh, yes, I've mistaken here :)